table/block_based_table_reader.cc

```diff
@@ -2551,12 +2553,29 @@ void BlockBasedTableIterator<TBlockIter, TValue>::FindKeyBackward() {
   // code simplicity.
 }
 
+template <class TBlockIter, typename TValue>
+void BlockBasedTableIterator<TBlockIter, TValue>::CheckOutOfBound() {
+  if (read_options_.iterate_upper_bound != nullptr &&
+      block_iter_points_to_real_block_ && block_iter_.Valid()) {
+    Slice current_user_key = block_iter_.key();
+    if (key_includes_seq_) {
```

Comment:
I am concerned that too many variables make the code less readable. Can we extend InternalIteratorBase with a "Slice user_key()" function and let the index iterator who already know if its keys are user key or internal key do the conversion?

Reply:
done.